modules/tensor_mechanics/src/materials/FiniteStrainHyperElasticViscoPlastic.C

```diff
@@ -0,0 +1,403 @@
+#include "FiniteStrainHyperElasticViscoPlastic.h"
+
+template<>
+InputParameters validParams<FiniteStrainHyperElasticViscoPlastic>()
+{
+  InputParameters params = validParams<FiniteStrainMaterial>();
+  params.addParam<Real>("resid_abs_tol", 1e-10, "Absolute Tolerance for flow rate residual equation");
+  params.addParam<Real>("resid_rel_tol", 1e-6, "Relative Tolerance for flow rate residual equation");
+  params.addParam<unsigned int>("maxiters", 50, "Maximum iteration for flow rate update");
+  params.addParam<unsigned int>("max_substep_iteration", 1, "Maximum number of substep iteration");
+  params.addRequiredParam<UserObjectName>("stress_user_object", "User object that computes stress using elastic RightCauchyGreenTensor");
+  params.addRequiredParam<std::vector<UserObjectName> >("flow_rate_user_objects", "List of User object names that computes flow rate and derivatives");
```

Comment:
Can you add a params.addClassDescription("description...") line? We are trying to get better in file documentation.